File: BookingApp/frontend/src/app/shared/models/Revision.ts
import { RentingItem } from "./reservations/RentingItem";

export class Revision{
    public id : Number;
    public clientId : Number;
    public rentingItemId : Number;
    public grade : Number;
    public description : String;
    public rentingItem : RentingItem;
    public reservationId : Number;

    constructor(id : Number,clientId : Number, rentingItemId : Number,grade : Number,description : String, reservationId : Number){
        this.id = id;
        this.clientId = clientId;
        this.rentingItemId = rentingItemId;
        this.grade = grade;
        this.description = description;
        this.reservationId = reservationId;
    }
}

File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/Revision.java
package com.example.BookingApp.reservations.model;

import com.example.BookingApp.renting.model.RentingItem;
import com.example.BookingApp.users.model.Client;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.GenericGenerator;

import javax.persistence.*;

@Entity
@Table(name="revisions")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Revision {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO, generator="seq")
    @GenericGenerator(name = "seq", strategy="increment")
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.DETACH)
    private Client client;
    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private RentingItem rentingItem;
    private double grade;
    private String description;
    private boolean approved;
    private Long reservationId;
}
